.github/workflows/release.yml

```diff
@@ -0,0 +1,101 @@
+name: "Release Pipeline"
+on:
+  push:
+    tags:
+      - v*.*.*
+
+  workflow_dispatch:
+
+jobs:
+  release-draft:
+    name: 'Draft release'
+    runs-on: ubuntu-latest
+    steps:
+        - name: "Create draft release"
+          uses: softprops/action-gh-release@v1
+          with:
+            draft: True
+
+  package-ubuntu:
+    needs: release-draft
+    name: 'Package Ubuntu ${{ matrix.UBUNTU }}'
+    runs-on: ubuntu-latest
+    container: 'ubuntu:${{ matrix.UBUNTU }}'
+    strategy:
+      fail-fast: true
+      matrix:
+        UBUNTU: ["18.04", "20.04", "20.10", "21.04"]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Generate build directory
+        run: mkdir -p build
+      - name: Install common dependencies
+        env:
+          DEBIAN_FRONTEND: noninteractive
+          TZ: Europe/Berlin
+        run: |
+          apt update
+          apt -y upgrade
+          apt -y install git build-essential lsb-release cmake libeigen3-dev libxml2-dev libboost-all-dev python3-dev python3-numpy
+      - name: Install PETSc
+        if: ${{ matrix.UBUNTU != '18.04' }}
+        env:
+          DEBIAN_FRONTEND: noninteractive
+        run: |
+          apt install -y petsc-dev
+      - name: Configure
+        working-directory: build
+        env:
+          PETSc: ${{ matrix.UBUNTU != '18.04' }}
+        run: |
+          cmake --version
+          cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Release -DCPACK_GENERATOR="DEB;TAR" -DPRECICE_PETScMapping=${{ env.PETSc }} -DBUILD_TESTING=OFF -DBUILD_SHARED_LIBS=ON ..
+      - name: Compile
+        working-directory: build
+        run: make -j $(nproc)
+      - name: Create packages
+        working-directory: build
+        run: cpack
+      - name: Upload debian package
+        uses: svenstaro/upload-release-action@v2
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          file: build/libprecice*.deb*
+          file_glob: true
+          tag: ${{ github.ref }}
+      - name: Upload tar package
+        uses: svenstaro/upload-release-action@v2
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          file: build/libprecice*.tar*
+          file_glob: true
+          tag: ${{ github.ref }}
```

Comment:
In the beginning, we added also the `.tar.gz` files as they can be useful to people that don't use Ubuntu, but something else with compatible dependencies. I have a hard time defining the exact target group and [who these few people can be](https://utilify.me/github-release-stats).

As the current list of packages is very long and can be confusing/daunting to the new user, what if we ommit the `.tar.gz` (and the respective hashes) in the next release, and find out if anyone complains?

The [example draft release](https://github.com/fsimonis/precice/releases/tag/v2.666.0) lists 18 different assets (including the code archives). Removing the `.tar.gz` would drop this to 10 files. Removing the 20.10, in addition, would drop this to 8 assets (three Debian packages with their hashes and the two standard code archives).

```suggestion
          cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Release -DCPACK_GENERATOR="DEB" -DPRECICE_PETScMapping=${{ env.PETSc }} -DBUILD_TESTING=OFF -DBUILD_SHARED_LIBS=ON ..
      - name: Compile
        working-directory: build
        run: make -j $(nproc)
      - name: Create packages
        working-directory: build
        run: cpack
      - name: Upload debian package
        uses: svenstaro/upload-release-action@v2
        with:
          repo_token: ${{ secrets.GITHUB_TOKEN }}
          file: build/libprecice*.deb*
          file_glob: true
          tag: ${{ github.ref }}
```

Reply:
Sure, let's try it out.